### DISCUSSION TITLE
Python installation fails without homebrew/core tap

### DISCUSSION THREAD

**Discussion by pacman-admin** [Homebrew/discussions#4937] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.22-55-gd68e3e5
ORIGIN: https://github.com/Homebrew/brew
HEAD: d68e3e563ef76b96b75df2e54686a94fec5c4474
Last commit: 2 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ec78efa59113a16f878b7ddcf59f0fcf39ed7b87
Core tap last commit: 12 minutes ago
Core tap branch: master
Core tap JSON: 30 Nov 16:13 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: quad-core 64-bit sandybridge
Clang: 10.0.0 build 1000
Git: 2.42.1 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
macOS: 10.13.6-x86_64
CLT: 10.1.0.0.1.1539992718
Xcode: N/A
```

### Output of `brew doctor`

```shell
Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  git

Warning: You are using macOS 10.13.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.


Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/cask) checkout master

Warning: Some installed formulae are missing dependencies.
You should `brew install` the missing dependencies:
  brew install curl openldap

Run `brew missing` for more details.
```

### Description of issue

I attempted to install python on a two 2011 iMacs running MacOS High Sierra.
The dependencies install without problems but the installation fails with this error:
`Error: An exception occurred within a child process:
  FormulaUnavailableError: No available formula with the name "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Aliases/python".`

**Comment by pacman-admin** [comment]
Running `brew tap --force homebrew/core` allows it to install.

A core tap should not be required to install a formula.

**Reply by Bo98** [reply]
What command are you running to get the error?

**Reply by pacman-admin** [reply]
Running `brew install python` or `brew install python3`

**Reply by Bo98** [reply]
Can you post the output of `brew install python --debug`? Should give more information.

**Reply by pacman-admin** [reply]
Sure.
I will post soon.

**Reply by pacman-admin** [reply]
Here is the output:`
Error: An exception occurred within a child process:
  FormulaUnavailableError: No available formula with the name "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Aliases/python".
Warning: Removed Sorbet lines from backtrace!
Rerun with --verbose to see the original backtrace
/usr/local/Homebrew/Library/Homebrew/formulary.rb:647:in `get_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:734:in `factory'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:126:in `block in load_formula_or_cask'
/usr/local/Homebrew/Library/Homebrew/api.rb:197:in `with_no_api_env_if_needed'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:118:in `load_formula_or_cask'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:83:in `block in to_formulae_and_casks'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:81:in `each'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:81:in `flat_map'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:81:in `to_formulae_and_casks'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:58:in `to_formulae'
/usr/local/Homebrew/Library/Homebrew/build.rb:226:in `<main>'
`

**Reply by pacman-admin** [reply]
Error: An exception occurred within a child process:
  FormulaUnavailableError: No available formula with the name "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Aliases/python".
/usr/local/Homebrew/Library/Homebrew/formulary.rb:647:in `get_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:734:in `factory'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/3.1.0/gems/sorbet-runtime-0.5.10461/lib/types/private/methods/call_validation.rb:157:in `bind_call'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/3.1.0/gems/sorbet-runtime-0.5.10461/lib/types/private/methods/call_validation.rb:157:in `validate_call'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/3.1.0/gems/sorbet-runtime-0.5.10461/lib/types/private/methods/call_validation.rb:89:in `block in create_validator_slow'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:126:in `block in load_formula_or_cask'
/usr/local/Homebrew/Library/Homebrew/api.rb:197:in `with_no_api_env_if_needed'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/3.1.0/gems/sorbet-runtime-0.5.10461/lib/types/private/methods/call_validation.rb:157:in `bind_call'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/3.1.0/gems/sorbet-runtime-0.5.10461/lib/types/private/methods/call_validation.rb:157:in `validate_call'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/3.1.0/gems/sorbet-runtime-0.5.10461/lib/types/private/methods/_methods.rb:270:in `block in _on_method_added'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:118:in `load_formula_or_cask'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:83:in `block in to_formulae_and_casks'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:81:in `each'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:81:in `flat_map'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:81:in `to_formulae_and_casks'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/3.1.0/gems/sorbet-runtime-0.5.10461/lib/types/private/methods/call_validation.rb:157:in `bind_call'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/3.1.0/gems/sorbet-runtime-0.5.10461/lib/types/private/methods/call_validation.rb:157:in `validate_call'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/3.1.0/gems/sorbet-runtime-0.5.10461/lib/types/private/methods/_methods.rb:270:in `block in _on_method_added'
/usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:58:in `to_formulae'
/usr/local/Homebrew/Library/Homebrew/build.rb:226:in `<main>'

**Reply by Bo98** [reply]
Thanks, I see the bug. I'll look into getting that fixed this weekend.

**Reply by Bo98** [reply]
Should be fixed now: https://github.com/Homebrew/brew/commit/27730f68c5e1671d62ddb76501c6a73c31c63a6e

**Reply by pacman-admin** [reply]
Thanks!

**Reply by pacman-admin** [reply]
The weird part is all other formulae install fine.

**Comment by pacman-admin** [comment]
And why is Python still at 3.11?
3.12 has been released.

**Reply by Bo98** [reply]
`brew install python@3.12` will install 3.12, though the default python3 exexutable is still 3.11 for now. You can read the caveats printed after installing 3.12 for instructions on how to change what `python3` points to.

**Reply by pacman-admin** [reply]
Also `brew install python@3.12` breaks for me.


I am using an old version of MacOS.

**Reply by Bo98** [reply]
What error do you get with 3.12?

**Reply by pacman-admin** [reply]
Working on it...

**Reply by pacman-admin** [reply]
It worked this time.
Very weird.
However, this is the second time and I previously installed python@3.11

**Comment by pacman-admin** [comment]
`Brew install python`: FAILS (Error)
`brew install python3`: FAILS (Same error as `Brew install python`)
`brew install python@3.12`: FAILS(probably because I am using an old OS)
`brew install python@3.11`: Unknown for now

**Comment by pacman-admin** [comment]
`brew install python@3.11` works perfectly.
I will investigate further.

**Reply by pacman-admin** [reply]
`brew install python3` works fine on Ubuntu.

(Technically, I installed it a month ago)
Kinda funny how brew works best on Linux because MacOS is never new enough.
